modules/programs/powerline-go.nix

```diff
@@ -0,0 +1,118 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.powerline-go;
+
+  # Convert an option value to a string to be passed as argument to
+  # powerline-go:
+  valueToString = value:
+    if builtins.isList value then
+      builtins.concatStringsSep "," (builtins.map valueToString value)
+    else if builtins.isAttrs value then
+      valueToString
+      (mapAttrsToList (key: val: "${valueToString key}=${valueToString val}")
+        value)
+    else
+      builtins.toString value;
+
+  modulesArgument = optionalString (cfg.modules != null)
+    "-modules ${valueToString cfg.modules}";
+
+  newlineArgument = optionalString cfg.newline "-newline";
+
+  pathAliasesArgument = optionalString (cfg.pathAliases != null)
+    "-path-aliases ${valueToString cfg.pathAliases}";
+
+  otherSettingPairArgument = name: value:
+    if value == true then "-${name}" else "-${name} ${valueToString value}";
+
+  otherSettingsArgument = optionalString (cfg.settings != { })
+    (concatStringsSep " "
+      (mapAttrsToList otherSettingPairArgument cfg.settings));
+
+  commandLineArguments = ''
+    ${modulesArgument} ${newlineArgument} ${pathAliasesArgument} ${otherSettingsArgument}
+  '';
+
+in {
+  meta.maintainers = [ maintainers.DamienCassou ];
+
+  options = {
+    programs.powerline-go = {
+      enable = mkEnableOption "Powerline-go";
+
+      modules = mkOption {
+        default = null;
+        type = types.nullOr (types.listOf types.str);
+        description = ''
+          List of module names to load (default
```

Comment:
Should we create it as a real default? so not default = null. This also would help of not needing to change it, if upstream changes this value.

Reply:
I would prefer to get rid of the comment rather than setting a non-null default. Having `null` as a default means "I want what upstream believes it's best" and I like it like that. I think it's a good default.

Reply:
That's also fine with me. Maybe a comment like the default is set by upstream (with a link to the settings)?

Reply:
Fixed! Thank you.